engine/render/src/render/material.cpp

```diff
@@ -113,7 +114,17 @@ namespace dmRender
                 m->m_NameHashToLocation.Put(name_hash, location);
 
                 HConstant render_constant = dmRender::NewConstant(name_hash);
-                SetConstantLocation(render_constant, location);
+                dmRender::SetConstantLocation(render_constant, location);
+
+                // Set correct size of the constant (Until the shader builder provides all the default values)
+                if (num_values > default_values_capacity)
+                {
+                    default_values_capacity = num_values;
+                    delete[] default_values;
+                    default_values = new dmVMath::Vector4[default_values_capacity];
+                    memset(default_values, 0, default_values_capacity * sizeof(dmVMath::Vector4));
+                }
+                dmRender::SetConstantValues(render_constant, default_values, num_values);
```

Comment:
Since we (previously) only set the values from the material, we only got a constant of that size.
But, we want to set default values, so that we can correctly check if the any index is out-of-bounds.